### ISSUE TITLE
AccessFrontend permission

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#2735]
Similar to the `AccessAdminPanel` permission have an `AccessFrontend` one to be able to lock down the frontend of the site to anonymous users.

This doesn't make sense in the headless CMS scenario, I'm not sure whether it would in the decoupled CMS one.

**Comment by sebastienros** [comment]
Permission, or feature of the content management module?

**Comment by sebastienros** [comment]
Should be part of the Admin feature then.

**Comment by Piedone** [comment]
Don't know enough of OC to comment on this.

**Comment by sebastienros** [comment]
I don't feel confident with this permission. Does it mean that anything that is not Admin would be prevented? 

When building a decoupled website (see video for reference), what about the custom pages? I assume we would just need the ItemController from content to be hidden. Also there aren't that many controllers that are visible on the front-end. I can only think about this one actually (not sure about the preview).

So in the end it might be better to hide the content item controller behind a feature. And if visible then the standard Content permissions would kick in. 

Other public controller:
- Diagnostics (404, 500)
- XmlRPC (more like an API)
- RSS

All of these are either already in custom features or could be moved in some.

**Comment by TFleury** [comment]
@Piedone removing 'View all content' permission from 'Anonymous' role is not sufficient ?

**Comment by Piedone** [comment]
No, because that's just for content items. This would be for all controllers, and if you revoke the AccessFrontend permission for a role than those users won't be able to access anything, content item or not, on the frontend, except for the actions/controlles with the `[AllowAnonymous]` attribute.

**Comment by sebastienros** [comment]
I think I get it now. Until I don't anymore. It would be a filter so there is nothing to do in any controller to handle it.

**Comment by Piedone** [comment]
I didn't say that, I'm saying that it would affect all controllers (without the controllers knowing anything). Again, I'm saying this would have exactly the same effect as in O1. I'm not sure whether a filter here would be the best approach too (it's one approach certainly), hooking into the authorization pipeline somewhere might be better.
